### ISSUE TITLE
Generated FX animators cause Gesture animator to break.

### ISSUE THREAD

**Issue by dotWAVE** [VRCFury/VRCFury#98]
When VRCFury adds layers to the FX animator, it should create the layers with an avatar mask to prevent breaking the gesture playable layer. Details on the behavior:
https://feedback.vrchat.com/sdk-bug-reports/p/sdk3-202010281558-cant-use-non-humanoid-animations-on-gesture-layer-unless-fx-la
https://ask.vrchat.com/t/gesture-layer-animations-not-playing/3624/11

**Comment by SenkyDragon** [comment]
It actually already does this. https://github.com/VRCFury/VRCFury/blob/main/com.vrcfury.vrcfury/Editor/VF/Feature/FixMasksBuilder.cs

**Comment by dotWAVE** [comment]
It does seem to copy the masks from the original animator to their respective layers in the generated animator. But it creates no FX mask for layers created by VRCF. 
![image](https://github.com/VRCFury/VRCFury/assets/2576219/d4833fa3-ccd5-4ab7-8842-c711216c535b)

**Comment by SenkyDragon** [comment]
It doesn't create a mask on that layer because it doesn't affect any transforms.

**Comment by SenkyDragon** [comment]
If you have a simple reproducible test case package that can showcase an issue, that would help in debugging any specific problem you are having.

**Comment by dotWAVE** [comment]
The specific issue is that unmasked layers in FX that use write defaults=off will break animations on non humanoid transforms in Gesture. If I use the feature to enforce Write Defaults=on, the states in the VF9 and VF4 FX layers change from WD off to WD on and non humanoid transforms in Gesture animate again. Additionally if I change the VF9 and VF4 layers to have WD=off and use the same mask that all of the other pictured layers have, non humanoid transforms in Gesture will also work.

**Comment by dotWAVE** [comment]
[VRCF-issue#98-example.zip](https://github.com/VRCFury/VRCFury/files/12443512/VRCF-issue.98-example.zip)
Import this into a project with VRCSDK, VRCFury, and Gesture Manager.

**Comment by SenkyDragon** [comment]
Using a mask on every FX layer isn't really an option, because it breaks material swaps on slots other than material slot 0. The only totally reliable solution is to either use WD on, or to move all non-humanoid bone transforms to FX (which is the solution I recommend 100% of the time).

**Comment by SenkyDragon** [comment]
We've been discussing this in discord for a long while actually (months? a year almost?) and constantly wind up back at the same solution: VRCFury should copy all layers from gesture into FX that involve transforms, remove the transforms from the gesture copy, and remove the humanoid bones from the fx copy. That way, we don't have to mask anything in FX and it all just works perfectly.

The only issue is that this is a relatively substantial change for many avatars, and the concern is that it's going to add even more to the complaints of "I USED VRCFURY TO ADD A SINGLE TOGGLE, WHY IS IT REDOING MY ENTIRE GESTURE CONTROLLER?"

**Comment by dotWAVE** [comment]
I see your point. I was under the impression that transforms in FX was poor practice so it wasn't something I considered. I'll give that a try.

**Comment by SenkyDragon** [comment]
They updated the creator docs a while back to suggest that it's acceptable, primarily due to this issue.
![image](https://github.com/VRCFury/VRCFury/assets/117542830/1cd8bb7c-c148-4197-a5dc-bb8ec72818eb)

Unfortunately, the wording is still unclear, because they also still say this, which I personally believe is incorrect:
![image](https://github.com/VRCFury/VRCFury/assets/117542830/cc597b68-bb0f-41cb-83c7-26bf64c2e0ff)

**Comment by Saberpeep** [comment]
**Preamble:**
Just ran into this issue trying to add one of Jerry's face tracking templates to an avatar designed for WD-Off. Someone really needs to do a proper deep dive and make some real documentation for all these animator quirks. There are however things I have found to be true based on my extensive experience **working with WD-Off**:
1. The FX animator can behave identically to the Gesture animator (to the point of being able to do everything on the FX layer), but not vice-versa. Animations on the FX animator will of course override animations on the Gesture animator.
2. Every layer both on the Gesture and FX animators must be masked other than the below exceptions.
3. Material swap animations that swap materials on meshes that have more than one material slot must not have a mask, _(this appears to be a unity bug)_. Removing this mask has no adverse effects if the animation only animates the material slots. If the mesh has only one material slot the layer can be safely masked.
4. **The first layer in the Gesture and FX animators should have no animation states and should have a mask on it that is a combination of all other layer masks in that animator.** This is of utmost importance since without a mask the FX animator will override all animations in the Gesture animator causing it to entirely cease to function on non-humanoid transforms. And without a mask in the base layer of the Gesture animator it will not function in-game (though it will in-editor in play mode).
5. The FX mask should have the humanoid portion disabled and requires at least one transform path even if it is unchecked _(though ideal its hierarchy matches your avatar)_. This mask can be shared amongst all layers in the FX animator. If you are for some reason doing all your animations in the FX animator (including transforms), then the masks can be set up as if it were a Gesture animator instead.
6. This masking behavior has changed over time as the SDK has been updated with no official word from VRChat other than the haphazard updates to the docs. This in combination with the majority of users using WD-On makes documenting and discussing this behavior difficult. 

**The Problem:**
As it is currently, VRCFury (1.506.0) adds a layer at the very top of the FX layer `[VFxx] Defaults` that has no mask thus breaking rule 4 and causing the Gesture layer animations to be overridden.

**A Solution?**
A simple solution may be to keep the existing base layer in place in the existing FX animator, and simply place this new layer just below it thus keeping the layer order intact and not breaking rule 4. As for that layer's mask, applying the existing base layer's mask may suffice as long as it's WD-Off, since it should be a mask either masking out all transforms, or restricting it to the transforms used in that animator.
